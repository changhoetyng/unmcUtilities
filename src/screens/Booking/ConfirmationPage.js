import React, {Component} from 'react';
import {View, Text, SafeAreaView, StyleSheet, Dimensions, Button} from 'react-native';
import {theme} from '../../styles/ThemeColour';
import HeaderBookingPage from '../../components/headerBookingPage';

class ConfirmationPage extends Component {
  render() {
    const screenWidth = Dimensions.get('window').width;
    const screenHeight = Dimensions.get('window').height;
    return (
      <SafeAreaView style={{backgroundColor: theme.backgroundPrimary, flex: 1}}>
          <HeaderBookingPage
          title="Confirmation"
          description="Make sure everthing is correct before the booking."
        />
        <View>
          <Text style={[styles.titleStyle, {fontSize: screenWidth * 0.05}]}>
            Details
          </Text>
          <View style={[styles.cardStyle, {fontSize: screenWidth * 0.8}]}>
            <Text>Selected Court: Court 1</Text>
            <Text>Time: 9.00 a.m. - 10.00a.m.</Text>
            <Text>Date: Saturday, 14 November 2020</Text>
          </View>
        </View>
        <View style={{marginTop: 8}}>
          <Text style={[styles.titleStyle, {fontSize: screenWidth * 0.05}]}>
            Terms and Conditions
          </Text>
          <View style={[styles.cardStyle, {fontSize: screenWidth * 0.8}]}>
            <Text>Upon booking the facility you must adhere the terms and conditions and the rules by the school</Text>
          </View>
        </View>
        <View style={{width: '100%', position: 'absolute', bottom: 0}}>
          <Button
            title="Book"
            onPress={() => console.log('worm')}
          />
        </View>
      </SafeAreaView>
    );
  }
}

export default ConfirmationPage;

const styles = StyleSheet.create({
    cardStyle: {
      width: '100%',
      backgroundColor: 'white',
      padding: 20,
    },
    titleStyle: {
      fontWeight: 'bold',
      color: '#757575',
      marginLeft: 17,
      marginBottom: 10,
    },
  });