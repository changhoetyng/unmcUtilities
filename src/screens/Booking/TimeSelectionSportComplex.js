import React, {Component} from 'react';
import {
  View,
  Text,
  SafeAreaView,
  StyleSheet,
  Dimensions,
  ScrollView,
  Button,
} from 'react-native';
import HeaderBookingPage from '../../components/headerBookingPage';
import {theme} from '../../styles/ThemeColour';
import DateTimePicker from '@react-native-community/datetimepicker';
import moment from 'moment';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {api} from '../../api/api';
import {CheckBox} from 'react-native-elements';
import {connect} from 'react-redux';
import {setBooking} from '../../store/actions/booking';
import FullPageLoader from '../../hooks/FullPageLoader';


class TimeSelectionSportComplex extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {},
      showCalendar: false,
      isLoading: false
    };
  }

  async componentDidMount() {
    this.setState({isLoading: true})
    const selectedFacilityId = await this.props.selectedFacilityId;
    await api
      .get(`/student/showSelectedFacility/${selectedFacilityId}`)
      .then((res) => {
        this.setState({data: res.data.data});
      })
      .catch((err) => {
        console.log(err);
      });
      this.setState({isLoading: false})
  }

  onChangeDate(event, selectedDate) {
    if (event.type === 'dismissed') {
      this.setState({showCalendar: false});
    }

    if (event.type === 'set') {
      this.setState({
        showCalendar: false,
        selectedDate: moment(selectedDate).format('DD/MM/YYYY'),
      });
    }
  }

  async onPressTime(selectedDate, time,subName) {
    this.setState({isLoading: true})
    let studentId = ''
    try {
      await api
      .get(`student/getuser`)
      .then((res) => {
        studentId = res.data.user.studentId
      })
      .catch((err) => {
        console.log(err);
      });

    const setBooking = {
      facilityId: this.state.data.facilityId,
      date: this.state.selectedDate,
      time: time,
      studentId: studentId,
      subCategoryId: selectedDate.subCategoryId,
      venueName: this.state.data.facility,
      subCategoryName: subName
    }
    this.setState({isLoading: false})
    await this.props.setBooking(setBooking)
    this.props.navigation.navigate('ConfirmationPage')
    } catch(e) {
      console.log(e)
    }
    

    // await api
    //   .patch(`/sportcomplex/booked/${selectedFacilityId}`,{
    //     facilityId: this.state.data.facilityId,
    //     date: this.state.selectedDate,
    //     time: time
    //   })
    //   .then((res) => {
    //     this.setState({data: res.data.data});
    //   })
    //   .catch((err) => {
    //     console.log(err);
    //   });
  }

  renderAvailability() {
    const screenWidth = Dimensions.get('window').width;
    let selectedDate = null;
    if (this.state.data.data) {
      selectedDate = this.state.data.data.filter(
        (item) => item.date === this.state.selectedDate,
      );
    }

    

    if (selectedDate) {
      return (
        <ScrollView contentContainerStyle={{paddingBottom: 260}}>
          {selectedDate.map((date, index) => {
            const subCatName = this.state.data.subCategory.find(
              (v) => v._id === date.subCategoryId,
            );
            return (
              <View key={index}>
                <View style={{width: '90%', alignSelf: 'center'}}>
                  <Text style={{fontSize: screenWidth * 0.055}}>
                    {subCatName.subName}
                  </Text>
                </View>
                <View
                  style={{
                    width: '90%',
                    alignSelf: 'center',
                    backgroundColor: 'white',
                    borderRadius: 10,
                  }}>
                  <View
                    style={{
                      justifyContent: 'space-between',
                      flexDirection: 'row',
                      flexWrap: 'wrap',
                      alignItems: 'flex-start',
                      padding: 10,
                    }}>
                    {date.timeListing.map((time, index) => {
                      if (time.timeStatus.status === 'open') {
                        return (
                          <View
                            key={index}
                            style={{
                              width: '40%',
                              padding: 5,
                              borderColor: theme.greyOne,
                              borderWidth: 2.5,
                              marginBottom: 8,
                            }}>
                            <TouchableOpacity onPress={() => this.onPressTime(date,time.time,subCatName.subName)}>
                              <Text
                                style={{fontWeight: 'bold', fontSize: 20}}
                                key={index}>
                                {time.time}
                              </Text>
                            </TouchableOpacity>
                          </View>
                        );
                      }
                    })}
                  </View>
                </View>
              </View>
            );
          })}
        </ScrollView>
      );
    }
  }

  render() {
    const screenWidth = Dimensions.get('window').width;
    const screenHeight = Dimensions.get('window').height;
    return (
      <View style={{flex: 1, backgroundColor: theme.backgroundPrimary}}>
        {this.state.isLoading && <FullPageLoader />}
        <HeaderBookingPage
          title="Select a time"
          description="Pick a time and date that's available for the court."
          backButton={() => this.props.navigation.goBack()}
        />
        <View style={{padding: 15}}>
          <Text
            style={{
              fontWeight: 'bold',
              fontSize: screenWidth * 0.055,
              marginBottom: 5,
            }}>
            Pick a Date:
          </Text>
          <TouchableOpacity
            style={{
              width: '40%',
              padding: 5,
              borderColor: theme.greyOne,
              borderWidth: 2.5,
              marginBottom: 8,
            }}
            onPress={() => this.setState({showCalendar: true})}>
            <Text style={{fontSize: screenWidth * 0.045, alignSelf: 'center'}}>
              {this.state.selectedDate
                ? this.state.selectedDate
                : moment().format('DD/MM/YYYY')}
            </Text>
          </TouchableOpacity>
          <Text
            style={{
              fontWeight: 'bold',
              fontSize: screenWidth * 0.055,
              marginBottom: 8,
            }}>
            Availability
          </Text>
          {this.renderAvailability()}
        </View>
        {this.state.showCalendar && (
          <DateTimePicker
            mode="date"
            value={new Date()}
            // minimumDate={new Date()}
            onChange={(event, selectedDate) =>
              this.onChangeDate(event, selectedDate)
            }
          />
        )}
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    selectedFacilityId: state.bookingReducer.selectedFacilityId,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setBooking: (bookInfo) => dispatch(setBooking(bookInfo)),
  };
};

export default connect(mapStateToProps,mapDispatchToProps)(TimeSelectionSportComplex);
