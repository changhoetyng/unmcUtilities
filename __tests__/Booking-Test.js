import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import ConfirmationPage from '../src/screens/Booking/ConfirmationPage';
import TimeSelectionRoom from '../src/screens/Booking/TimeSelectionRoom';
import TimeSelectionSportComplex from '../src/screens/Booking/TimeSelectionSportComplex';
import VenueSelection from '../src/screens/Booking/VenueSelection'
import renderer from 'react-test-renderer';
import mock from 'react-native-permissions/mock';
const middlewares = [];
const mockStore = configureStore(middlewares);

jest.mock('@react-native-community/async-storage', () =>
  require('@react-native-community/async-storage/jest/async-storage-mock'),
);

jest.mock('react-native-permissions', () => {
  return mock;
});

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  pop: jest.fn(),
  dispatch: jest.fn(),
};

describe('Confirmation Screen', () => {
  jest.useFakeTimers();

  const store = mockStore({bookingReducer: {}});
  const confirmationPage = renderer.create(
    <Provider store={store}>
      <ConfirmationPage navigation={navigation} />
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Render Check In Page', () => {
    expect(confirmationPage.toJSON()).toMatchSnapshot();
  });
});

describe('TimeSelectionRoom Screen', () => {
  jest.useFakeTimers();

  const store = mockStore({bookingReducer: {}});
  const timeSelectionRoom = renderer.create(
    <Provider store={store}>
      <TimeSelectionRoom navigation={navigation} />
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Render TimeSelectionRoom Page',async () => {
    expect(timeSelectionRoom.toJSON()).toMatchSnapshot();
  });
});

describe('TimeSelectionSportComplex Screen', () => {
  jest.useFakeTimers();

  const store = mockStore({bookingReducer: {}});
  const timeSelectionSportComplex = renderer.create(
    <Provider store={store}>
      <TimeSelectionSportComplex navigation={navigation} />
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Render TimeSelectionSportComplex Page', async () => {
    expect(timeSelectionSportComplex.toJSON()).toMatchSnapshot();
  });
});

describe('TimeSelectionSportComplex Screen', () => {
  jest.useFakeTimers();

  const store = mockStore({bookingReducer: {}});
  const timeSelectionSportComplex = renderer.create(
    <Provider store={store}>
      <TimeSelectionSportComplex navigation={navigation} />
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Render TimeSelectionSportComplex Page', async () => {
    expect(timeSelectionSportComplex.toJSON()).toMatchSnapshot();
  });
});

describe('VenueSelection Screen', () => {
  jest.useFakeTimers();

  const store = mockStore({bookingReducer: {}});
  const venueSelection = renderer.create(
    <Provider store={store}>
      <VenueSelection navigation={navigation} />
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Render VenueSelection Page', async () => {
    expect(venueSelection.toJSON()).toMatchSnapshot();
  });
});