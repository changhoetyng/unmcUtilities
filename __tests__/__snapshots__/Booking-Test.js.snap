// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Screen Render Check In Page 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  style={
    Object {
      "zIndex": 1100,
    }
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
        "zIndex": 1000,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "display": "flex",
          "height": 150,
          "justifyContent": "space-around",
          "width": 150,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/loading.gif",
          }
        }
        style={
          Object {
            "height": "70%",
            "width": "70%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Loading
      </Text>
    </View>
  </View>
</Modal>
`;

exports[`TimeSelectionRoom Screen Render TimeSelectionRoom Page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#DEE4E8",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "zIndex": 1100,
      }
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "zIndex": 1000,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "display": "flex",
            "height": 150,
            "justifyContent": "space-around",
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/loading.gif",
            }
          }
          style={
            Object {
              "height": "70%",
              "width": "70%",
            }
          }
        />
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Loading
        </Text>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "backgroundColor": "#DEE4E8",
        "padding": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 60,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontWeight": "bold",
            "marginLeft": 8,
          },
          Object {
            "fontSize": 45,
          },
        ]
      }
    >
      Select a time
    </Text>
    <Text
      style={
        Object {
          "color": "#70A9FF",
          "fontSize": 26.250000000000004,
          "fontWeight": "bold",
          "marginLeft": 8,
          "width": "50%",
        }
      }
    >
      Pick a time and date that's available for the room.
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 41.25,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Pick a Date:
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "borderColor": "#757575",
            "borderWidth": 2.5,
            "marginBottom": 8,
            "opacity": 1,
            "padding": 5,
            "width": "40%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 33.75,
            }
          }
        >
          27/04/2021
        </Text>
      </View>
    </RNGestureHandlerButton>
    <Text
      style={
        Object {
          "fontSize": 41.25,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      Availability
    </Text>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "flexWrap": "wrap",
            "height": 525,
            "padding": 10,
            "width": 525,
          }
        }
      >
        <View
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 37.5,
              }
            }
          >
            Select Duration
          </Text>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              Object {
                "checked": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#ededed",
                "borderRadius": 3,
                "borderWidth": 1,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#2089dc",
                      "fontSize": 24,
                    },
                    Object {
                      "minWidth": 24,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#43484d",
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                1 Hours
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              Object {
                "checked": false,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#ededed",
                "borderRadius": 3,
                "borderWidth": 1,
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#bfbfbf",
                      "fontSize": 24,
                    },
                    Object {
                      "minWidth": 24,
                    },
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#43484d",
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                testID="checkboxTitle"
              >
                2 Hours
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "bottom": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Done
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`TimeSelectionSportComplex Screen Render TimeSelectionSportComplex Page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#DEE4E8",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "zIndex": 1100,
      }
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "zIndex": 1000,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "display": "flex",
            "height": 150,
            "justifyContent": "space-around",
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/loading.gif",
            }
          }
          style={
            Object {
              "height": "70%",
              "width": "70%",
            }
          }
        />
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Loading
        </Text>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "backgroundColor": "#DEE4E8",
        "padding": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 60,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontWeight": "bold",
            "marginLeft": 8,
          },
          Object {
            "fontSize": 45,
          },
        ]
      }
    >
      Select a time
    </Text>
    <Text
      style={
        Object {
          "color": "#70A9FF",
          "fontSize": 26.250000000000004,
          "fontWeight": "bold",
          "marginLeft": 8,
          "width": "50%",
        }
      }
    >
      Pick a time and date that's available for the court.
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 41.25,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Pick a Date:
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "borderColor": "#757575",
            "borderWidth": 2.5,
            "marginBottom": 8,
            "opacity": 1,
            "padding": 5,
            "width": "40%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 33.75,
            }
          }
        >
          27/04/2021
        </Text>
      </View>
    </RNGestureHandlerButton>
    <Text
      style={
        Object {
          "fontSize": 41.25,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      Availability
    </Text>
  </View>
</View>
`;

exports[`TimeSelectionSportComplex Screen Render TimeSelectionSportComplex Page 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#DEE4E8",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "zIndex": 1100,
      }
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "zIndex": 1000,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "display": "flex",
            "height": 150,
            "justifyContent": "space-around",
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/loading.gif",
            }
          }
          style={
            Object {
              "height": "70%",
              "width": "70%",
            }
          }
        />
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Loading
        </Text>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "backgroundColor": "#DEE4E8",
        "padding": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 60,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontWeight": "bold",
            "marginLeft": 8,
          },
          Object {
            "fontSize": 45,
          },
        ]
      }
    >
      Select a time
    </Text>
    <Text
      style={
        Object {
          "color": "#70A9FF",
          "fontSize": 26.250000000000004,
          "fontWeight": "bold",
          "marginLeft": 8,
          "width": "50%",
        }
      }
    >
      Pick a time and date that's available for the court.
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 41.25,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Pick a Date:
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "borderColor": "#757575",
            "borderWidth": 2.5,
            "marginBottom": 8,
            "opacity": 1,
            "padding": 5,
            "width": "40%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 33.75,
            }
          }
        >
          27/04/2021
        </Text>
      </View>
    </RNGestureHandlerButton>
    <Text
      style={
        Object {
          "fontSize": 41.25,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      Availability
    </Text>
  </View>
</View>
`;

exports[`VenueSelection Screen Render VenueSelection Page 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#DEE4E8",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "zIndex": 1100,
      }
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "zIndex": 1000,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "display": "flex",
            "height": 150,
            "justifyContent": "space-around",
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/loading.gif",
            }
          }
          style={
            Object {
              "height": "70%",
              "width": "70%",
            }
          }
        />
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Loading
        </Text>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "backgroundColor": "#DEE4E8",
        "padding": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 60,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontWeight": "bold",
            "marginLeft": 8,
          },
          Object {
            "fontSize": 45,
          },
        ]
      }
    >
      Choose a Room
    </Text>
    <Text
      style={
        Object {
          "color": "#70A9FF",
          "fontSize": 26.250000000000004,
          "fontWeight": "bold",
          "marginLeft": 8,
          "width": "50%",
        }
      }
    >
      Pick a room and check for it's availability.
    </Text>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "width": "90%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
